### PR TITLE
fix types of node crypto module

### PR DESCRIPTION
**Pull request by plantain-00**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://nodejs.org/dist/latest-v6.x/docs/api/crypto.html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by vvakame** [comment]
related #11026
@plantain-00 please rebase to master/HEAD.

**Comment by plantain-00** [comment]
I think the last commit fixed the conflicts of ECDH interface and createECDH function.

File: node/node.d.ts [hunk 1]
```diff
@@ -2189,63 +2201,86 @@ declare module "crypto" {
...
+        sign(private_key: string | { key: string; passphrase: string }, output_format: "latin1" | "hex" | "base64"): string;
     }
     export function createVerify(algorith: string): Verify;
     export interface Verify extends NodeJS.WritableStream {
-        update(data: any): void;
+        update(data: string | Buffer): Verify;
+        update(data: string | Buffer, input_encoding: "utf8" | "ascii" | "latin1"): Verify;
         verify(object: string, signature: Buffer): boolean;
-        verify(object: string, signature: string, signature_format: 'latin1' | 'hex' | 'base64'): boolean;
+        verify(object: string, signature: string, signature_format: string): boolean;
```
Comment (zhengbli) [inline node/node.d.ts]: Why the type of `signature_format` was reduced to `string`? the doc still says it is 'latin1', 'hex' or 'base64'

File: node/node.d.ts [hunk 2]
```diff
@@ -2189,63 +2201,86 @@ declare module "crypto" {
...
-        getPrivateKey(encoding?: string): string;
-        setPublicKey(public_key: string, encoding?: string): void;
-        setPrivateKey(public_key: string, encoding?: string): void;
+        generateKeys(): Buffer;
+        generateKeys(encoding: "latin1" | "hex" | "base64"): string;
+        computeSecret(other_public_key: Buffer): Buffer;
+        computeSecret(other_public_key: string, input_encoding: "latin1" | "hex" | "base64"): Buffer;
+        computeSecret(other_public_key: string, input_encoding: "latin1" | "hex" | "base64", output_encoding: "latin1" | "hex" | "base64"): string;
+        getPrime(): Buffer;
+        getPrime(encoding: "latin1" | "hex" | "base64"): string;
```
Comment (zhengbli) [inline node/node.d.ts]: The type `"latin1" | "hex" | "base64"` and the type `"utf8" | "ascii" | "latin1"` are used in a lot of places. Can you make an type alias for each and use that instead? So you won't have typos.
```
type InputEncoding = "latin1" | "hex" | "base64";
... input_encoding: InputEncoding
```

**Comment by plantain-00** [comment]
The type of `verify` in `Verify` is fixed
and the format type is irregular, so I name it as its type names:
```ts
    type Utf8AsciiLatin1Encoding = "utf8" | "ascii" | "latin1";
    type HexBase64Latin1Encoding = "latin1" | "hex" | "base64";
    type Utf8AsciiBinaryEncoding = "utf8" | "ascii" | "binary";
    type HexBase64BinaryEncoding = "binary" | "base64" | "hex";
    type ECDHKeyFormat = "compressed" | "uncompressed" | "hybrid";
```

**Comment by vvakame** [comment]
LGTM.
cc/ @zhengbli @hriss95

**Comment by zhengbli** [comment]
👍

**Pull request merged**